protocols/gossipsub/src/protocol.rs

```diff
@@ -634,4 +689,241 @@ mod tests {
         assert_eq!(protocol_config.protocol_ids[0].protocol, "/foosub");
         assert_eq!(protocol_config.protocol_ids[1].protocol, "/floodsub/1.0.0");
     }
+
+    #[test]
```

Comment:
sorry I was probably not clear, we want functional tests, test that when passing a config with max size per topics to the `Behavior` it is passed and respected at the protocol level, they should live in `behavior/tests.rs` and use the `inject_nodes` api

Reply:
yeah but also test that when we receive a message from a peer that doesn't limit message size per topic, we get `ValidationError::MessageSizeTooLargeForTopic` on the `invalid_messages`

Reply:
friendly ping @elijahhampton

Reply:
yeah was also referring to the conflicts